src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs

```diff
@@ -41,15 +57,28 @@ private bool IsValidParameterList(IReadOnlyList<KeyValuePair<string, object>> pa
                     return false;
                 }
 
-                if (paramPair.Key == NLogLogger.OriginalFormatPropertyName)
+                char firstChar = paramPair.Key[0];
+                if (!char.IsDigit(firstChar))
                 {
-                    if (originalMessageIndex.HasValue)
+                    if (firstChar == '@' || firstChar == '$')
                     {
-                        originalMessageIndex = null;
-                        return false;
+                        customCaptureTypes = true;
+                        isPositional = false;
                     }
+                    else if (firstChar == '{' && paramPair.Key == NLogLogger.OriginalFormatPropertyName)
+                    {
+                        if (originalMessageIndex.HasValue)
+                        {
+                            originalMessageIndex = null;
+                            return false;
+                        }
 
-                    originalMessageIndex = i;
+                        originalMessageIndex = i;
+                    }
+                    else
+                    {
+                        isPositional = false;
+                    }
```

Comment:
I think line 68-81 is duplicated in another file. Please deduplicate

Reply:
Fixed